Add spec for governance token staking from general account and locked rewards counting towards staking balance (#2350)
vegaprotocol/specs: protocol/0056-REWA-rewards_overview.md

Run for another epoch with no fee generated. Expect no transfer to be made to the reward pools of the accounts.

### Locked Reward

> When a party earns rewards (in governance token), they are transferred into the party's locked account and locked for the appropriate period before vesting. Sum of rewards in vesting account and locked account will be reflected in the party's "LOCKED_FOR_STAKING" account and they can be staked.

Reviewer: ```suggestion
When a party earns locked rewards (in governance token), they are transferred into the party's locked rewards account and locked for the appropriate period before vesting.
Sum of rewards in vesting account and locked account should count toward the party's staking balance and they can be [staked](0059-STKG-simple_staking_and_delegating.md).
```

> When a party earns locked rewards (in governance token), they are transferred into the party's locked rewards account and locked for the appropriate period before vesting.
> Sum of rewards in vesting account and locked account should count toward the party's staking balance and they can be [staked](0059-STKG-simple_staking_and_delegating.md).

Reviewer 2: Small clarification, in cores implementation there are no separate locked rewards and vesting accounts, they are one single account.
```suggestion
When a party earns locked rewards (in governance token), they are transferred into the party's vesting account account but locked for the appropriate period before vesting.
The rewards in the vesting account (both locked and unlocked) should count toward the party's staking balance and they can be [staked](0059-STKG-simple_staking_and_delegating.md).
```

> - When a party receives a governance token reward that is locked, the [Locked_FOR_REWARD] balance for the party is incremented by the corresponding amount. (<a name="0056-REWA-200" href="#0056-REWA-200">0056-REWA-200</a>).

Reviewer 2: ```suggestion
- When a party receives a governance token reward that is locked, the [LOCKED_FOR_STAKING] balance for the party is incremented by the corresponding amount. (<a name="0056-REWA-200" href="#0056-REWA-200">0056-REWA-200</a>).
```

> - When a party transfers locked tokens out of the [Locked_FOR_REWARD] account, the balance for the party is decreased by the corresponding amount. (<a name="0056-REWA-201" href="#0056-REWA-201">0056-REWA-201</a>).

Reviewer: ```suggestion
```

Author: why delete this AC?

> - Balance of the governance asset in the [Locked_FOR_REWARD] account counts towards the party's staking balance and can be nominated to a validator and earn staking rewards. (<a name="0056-REWA-202" href="#0056-REWA-202">0056-REWA-202</a>).

Reviewer 2: ```suggestion
- Total balance of the governance asset in the relevant vesting account counts towards the party's staking balance and can be nominated to a validator and earn staking rewards. (<a name="0056-REWA-202" href="#0056-REWA-202">0056-REWA-202</a>).
```

> - A party can stake the assets in the [Locked_FOR_REWARD] account. (<a name="0056-REWA-202" href="#0056-REWA-202">0056-REWA-202</a>).

Reviewer: ```suggestion
- Balance of the governance asset in the [Locked_FOR_REWARD] account counts towards the party's staking balance and can be nominated to a validator and earn staking rewards. (<a name="0056-REWA-202" href="#0056-REWA-202">0056-REWA-202</a>).
```
